src/test/java/CuidTest.java

```diff
@@ -13,13 +13,29 @@ public void createCuid() throws Exception {
         assertEquals(Cuid.createCuid().charAt(0), 'c');
     }
 
+    @Test
+    public void testForCollisions() {
+        assertEquals(hasNoCollisions(), true);
+    }
+
+    @Test
+    public void shouldValidateACuid() throws Exception {
+        final String cuid = Cuid.createCuid();
+        assertTrue(Cuid.validate(cuid));
+    }
+
+    @Test
+    public void shouldNotAcceptNullCuidWhenValidate() throws Exception {
+        assertFalse(Cuid.validate(null));
+    }
+
     private boolean hasNoCollisions() {
         int iterations = 4200000;
         Map<String, String> cuids = new HashMap<>();
 
         for (int i = 0; i < iterations; ++i) {
             String cuid = Cuid.createCuid();
-            if (cuids.containsKey(cuid)) {
+            if (cuids.containsKey(cuid) && Cuid.validate(cuid)) {
```

Comment:
Yeah I agree with you, actually this guy just are there to make sure it's valid in '4200000' iteractions. I'll remove that.